src/Plugin/Visualisation/Style/AxisChart.php

```diff
@@ -867,13 +867,13 @@ protected function chartBuildSettings(array $records) {
     // Check if values are auto.
     $tick_values_field = $this->config('axis', 'x', 'tick', 'values', 'field');
 
-    if ($settings['axis']['x']['type'] === '') {
+    if ($settings['axis']['x']['type'] === '' && !empty($record)) {
```

Comment:
If there is no record, maybe this should bomb out a lot earlier? eg at the top of this method
```
if (!$record) {
  drupal_set_message('Invalid record', 'error');
  return [];
}
```